pkg/storage/fs/s3ng/blobstore/blobstore.go

```diff
@@ -71,9 +71,9 @@ func (bs *Blobstore) Upload(node *node.Node, source string) error {
 	}
 	defer reader.Close()
 
-	_, err1 := bs.client.PutObject(context.Background(), bs.bucket, bs.path(node), reader, node.Blobsize, minio.PutObjectOptions{ContentType: "application/octet-stream"})
+	_, err = bs.client.PutObject(context.Background(), bs.bucket, bs.path(node), reader, node.Blobsize, minio.PutObjectOptions{ContentType: "application/octet-stream", SendContentMd5: true})
```

Comment:
This might be a performance problem. `SendContentMd5: true` causes minio-go to reread the whole file before making the actual putobject request as it needs to send a `Content-MD56` header. Maybe a more recent version of minio-go supports sending that as a trailer.

Reply:
nah ... we can't send our MD5 with this lib ... the `SendContentMd5` flag will always calculate the hash itself and there is no way to sneak in a `Content-MD5` header.

Reply:
let's try to get a fix into upstream: https://github.com/minio/minio-go/pull/1867